### PR TITLE
Add v1 prefix to routes

### PR DESCRIPTION
**Pull request by Ellenn-A**
# Changes 
- added v1 prefix to routes - this was done through updating file structure - adding 'v1' to routes 
- removed v1 prefix from /api-docs and /swagger

# Why we are doing this
- what we are doing is moving the prefix v1 to be part of routes not the baseUrl so that when we do the [openapi-merger](https://digicatapult.atlassian.net/jira/software/projects/L3/boards/152?selectedIssue=L3-181) we do not have a mismatch in the routes.  We found out that matchmaker actually had the v1 as part of routes so we just need to unify them.  I am including a screenshot of the merged spec and what the issue looked like where you can see the prefixes were mismatched Screenshot 2024-04-08 at 14 45 06

### Tickets: 
- https://digicatapult.atlassian.net/jira/software/projects/L3/boards/152?selectedIssue=L3-241
- https://digicatapult.atlassian.net/jira/software/projects/L3/boards/152?selectedIssue=L3-242

### CODE REVIEW

File: app/server.js [hunk 1]
```diff
@@ -78,6 +78,8 @@ export async function createHttpServer() {
       : {}
 
   v1ApiDoc.info.title = API_SWAGGER_HEADING
+  // console.log('fails after this ')
+  // debugger
```
Comment (mattdean-digicatapult) [inline app/server.js:82]: need to remove debugging logs

File: app/server.js [hunk 2]
```diff
@@ -86,9 +88,11 @@ export async function createHttpServer() {
     dependencies: {
       apiService: v1ApiService,
     },
-    paths: [path.resolve(__dirname, `api-${API_MAJOR_VERSION}/routes`)],
+    paths: [path.resolve(__dirname, `routes`)],
   })
 
+  // console.log('fails before this ')
```
Comment (mattdean-digicatapult) [inline app/server.js:94]: remove debugging logs

File: test/helper/routeHelper.js [hunk 3]
```diff
@@ -7,7 +7,7 @@ const { API_MAJOR_VERSION } = env
 
 export async function apiDocs({ app }) {
   return request(app)
-    .get(`/${API_MAJOR_VERSION}/api-docs`)
+    .get(`/api-docs`)
```
Comment (mattdean-digicatapult) [inline test/helper/routeHelper.js:10]: this is inconsistent with how we do this elsewhere. For exmaple on matchmaker (https://github.com/digicatapult/sqnc-matchmaker-api/blob/7a32be0c5fd8ab07a8b68201b9307fd402ed7943/src/server.ts) we do 

```js
app.get('/api-docs', (_req, res) => res.json(swaggerJson))
```

**Review by mattdean-digicatapult — COMMENTED** [review]
So as far as I can tell this doesn't change anything in the exposed api? It looks like you've just moved where the `v1` prefix is implemented on the routes. Can I ask why we're doing this as I'm likely missing some context

**Review by n3op2 — COMMENTED** [review]
apart from Matt's comments - this LGTM

File: test/helper/routeHelper.js (hunk 3 above)
Comment (Ellenn-A) [reply test/helper/routeHelper.js:10]: hmm the motivation was to do the same that matchmaker does like here: https://github.com/digicatapult/sqnc-matchmaker-api/blob/7a32be0c5fd8ab07a8b68201b9307fd402ed7943/src/server.ts#L33

**Review by mattdean-digicatapult — APPROVED** [review]

**Review by msterle — APPROVED** [review]
lgtm

**Comment by msterle** [comment]
> So as far as I can tell this doesn't change anything in the exposed api? It looks like you've just moved where the `v1` prefix is implemented on the routes. Can I ask why we're doing this as I'm likely missing some context

We are bringing into line all the sqnc services so that we can share a common ingress rule.

As it stood, we had servers with the /v1 prefix on some services, /v1 prefixes on `api-docs` on occasion, and sometimes something else entirely, which did not work with openapie-merger.

The standard we are working towards for this ingress is:
- swagger spec servers do not have version prefixes, as a server should be able to serve all versions
- routes have version prefixes, such that they can be versioned later, if needed
- /api-docs and /swagger should not be version prefixed, as they should be the source of truth for all existing and future versions

**Pull request merged**